llvm/lib/Target/AArch64/AArch64Combine.td

```diff
@@ -140,6 +140,15 @@ def fullrev: GICombineRule <
   (apply [{ applyFullRev(*${root}, MRI); }])
 >;
 
+def fullrevpoison: GICombineRule <
+  (defs root:$root, shuffle_matchdata:$matchinfo),
+  (match (G_POISON $src2),
+         (G_SHUFFLE_VECTOR $src, $src1, $src2, $mask):$root,
+         [{ return ShuffleVectorInst::isReverseMask(${mask}.getShuffleMask(),
+                                                    ${mask}.getShuffleMask().size()); }]),
+  (apply [{ applyFullRev(*${root}, MRI); }])
+>;
```

Comment:
You can repeat the existing pattern for the opcodes with foreach, e.g. see idempotent_prop_frags in the generic combines

Reply:
Thanks for the tip! Done!